dsa-23au/java-dsa/hash-map/src/main/java/dev/codewithfriends/HashMapWrapper.java

```diff
@@ -80,20 +80,25 @@ public HashMapWrapper(int m, int maxSize) {
         this.itemCount = 0;
     }
 
-    private static int hash(K key) {
+    public int hash(K key) {
         return (((key.hashCode() * this.a) + this.b) % this.m);
     }
 
-    public V get(K key) {
-        int index = hash(key);
-
-        if (_value_array[index] != null) {
-            List<Map.Entry<K,V>> matches =_value_array[index].stream().filter((Map.Entry<K,V> e) -> e.getKey().equals(key)).collect(Collectors.toList());
-            if (matches.size() > 0) {
-                return matches.get(0).getValue();
+    @Override
+    public V get(Object key) {
+        try {
+            int index = hash((K)key);
+
+            if (_value_array[index] != null) {
+                List<Map.Entry<K,V>> matches =_value_array[index].stream().filter((Map.Entry<K,V> e) -> e.getKey().equals(key)).collect(Collectors.toList());
```

Comment:
"List<Map.Entry<K,V>> matches =_value_array[index].stream().filter((Map.Entry<K,V> e) -> e.getKey().equals(key)).collect(Collectors.toList());"

What does the -> operator do?

Reply:
The `->` operator is used in the lamba notation, an anonymous function, to separate parameters from the return value. In this case:

``
(Map.Entry<K,V> e) -> e.getKey().equals(key)
```

is a lambda function that takes a map entry and returns a boolean which is true when the entry key equals a given `key` and false otherwise.

The `.filter` operation takes such a lambda (Entry -> boolean)  to "filter" out and keep all the entries that evaluate to true. Functions that are used in this way, and map arbitrary types to boolean, are also called "predicates" from first-order logic.